cis194/week1/daniel-deng/src/CIS194/Homework01/Exercise05.hs

```diff
@@ -0,0 +1,22 @@
+module CIS194.Homework01.Exercise05 (hanoi) where
+
+type Disks = Integer
+type Peg   = String
+type Move  = (Peg, Peg)
+
+{-
+  To move n discs (stacked in increasing size) from peg a to peg b
+  using peg c as temporary storage,
+  1. move n − 1 discs from a to c using b as temporary storage
+  2. move the top disc from a to b
+  3. move n − 1 discs from c to b using a as temporary storage.
+-}
+hanoi :: Disks -> Peg -> Peg -> Peg -> [Move]
+hanoi 0 _ _ _                  = []
+hanoi 1 start destination _    = [(start, destination)]
+hanoi n start destination temp
+  | 0 > n     = []
```

Comment:
should this be `n == 0`?

Reply:
Good catch @stevemao ! Merged to the first branch of guard.